src/main/java/com/opentable/versionedconfig/GitPropertiesFactoryBean.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.versionedconfig;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.opentable.spring.PropertySourceUtil;
+import com.opentable.spring.SpecializedConfigFactory;
+
+public class GitPropertiesFactoryBean implements FactoryBean<GitProperties> {
+
+    @Inject
+    private Optional<List<VersionedURICustomizer>> versionedURICustomizer = Optional.empty();
+
+    @Inject
+    private ConfigurableEnvironment environment;
+
+    private final String name;
+
+    public GitPropertiesFactoryBean(String name) {
+        Objects.requireNonNull(name);
+        if (name.contains(".")) {
+            throw new IllegalArgumentException("name cannot contain a period");
+        }
+        this.name = name.trim();
+    }
+
+    @Override
+    public GitProperties getObject() throws Exception {
+
+        String prefix = "ot.versioned-config." + name;
+        final SpecializedConfigFactory<VersionedSpringProperties> specializedConfigFactory = SpecializedConfigFactory.create(environment,
+                VersionedSpringProperties.class, "ot.versioned-config.${name}");
+        final VersionedSpringProperties versionedSpringProperties = specializedConfigFactory.getConfig(name);
+        if (versionedSpringProperties.getRemote().isEmpty()) {
+            throw new IllegalArgumentException("Must specify at least one uri");
+        }
+        if (StringUtils.isEmpty(versionedSpringProperties.getBranch())) {
+            throw new IllegalArgumentException("Must specify branch");
+        }
+        if (StringUtils.isEmpty(versionedSpringProperties.getLocal())) {
+            throw new IllegalArgumentException("Must specify local");
+        }
+        final Path localPath = Paths.get(versionedSpringProperties.getLocal());
+
+        /*
+         * ot.versioned-config.(name).local
+         * ot.versioned-config.(name).branch
+         * ot.versioned-config.(name).remote
+         * ot.versioned-config.(name).secrets.(index)
+         *
+         */
+        // Convert to mutable uris
```

Comment:
this could just be me.  Can we expand on the comment bit more, as of now I am not getting much out of it.